src/GafferScene/BranchCreator.cpp

```diff
@@ -131,6 +131,52 @@ void validateDestination( const ScenePlug::ScenePath &destination )
 	}
 }
 
+IECore::CompoundObjectPtr attributesAfterCommonRoot(
+	const ScenePlug::ScenePath &queryPath, const ScenePlug::ScenePath &refPath, const ScenePlug *inPlug
+)
+{
+	unsigned int matchingLength = std::min( queryPath.size(), refPath.size() );
+	for( unsigned int i = 0; i < matchingLength; i++ )
+	{
+		if( queryPath[ i ] != refPath[ i ] )
+		{
+			matchingLength = i;
+			break;
+		}
+	}
+
+	IECore::CompoundObjectPtr result;
+
+	ScenePlug::ScenePath curPath = queryPath;
+	curPath.resize( matchingLength + 1 );
+	for( unsigned int i = matchingLength; i < queryPath.size(); i++ )
```

Comment:
I've got a feeling this could be expressed a bitt more succinctly, doing the matching and the query in a single loop. Maybe something like this?

```
CompoundObjectPtr result;
ScenePath currentPath;
for( const auto &p : queryPath )
{
    currentPath.push_back( p );
    if( !result && currentPath.size() > refPath.size() || p != refPath[currentPath.size()-1] )
    {
       // We're past the common root
        result = new CompoundObject;
    }
    if( result )
    {
       // Query attributes and update result
    }
}
```